projects/schematics/src/shared/utils/lib-utils.ts

```diff
@@ -1047,3 +1026,95 @@ export function runExternalSpartacusLibrary(
 function createModuleFileName(config: FeatureConfig): string {
   return `${dasherize(config.moduleName)}-feature.module.ts`;
 }
+
+/**
+ * Used to sort the features in the correct order.
+ */
+function calculateSort(libraryA: string, libraryB: string): number {
+  const indexA = installationOrder.indexOf(libraryA);
+  const indexB = installationOrder.indexOf(libraryB);
+
+  /**
+   * In case a feature module is _not_ found in the `installationOrder`,
+   * we want to sort it at the end of the list.
+   */
+  return (indexA > -1 ? indexA : Infinity) - (indexB > -1 ? indexB : Infinity);
+}
+
+export function orderInstalledFeatures<T extends LibraryOptions>(
+  options: T
+): Rule {
+  return (tree: Tree, context: SchematicContext): void => {
+    context.logger.info('Ordering the installed Spartacus features...');
+
+    const basePath = process.cwd();
+    const { buildPaths } = getProjectTsConfigPaths(tree, options.project);
+    for (const tsconfigPath of buildPaths) {
+      const { appSourceFiles } = createProgram(tree, basePath, tsconfigPath);
+
+      let spartacusFeaturesModule: SourceFile | undefined;
```

Comment:
Maybe we can re-purpose `spartacusFeatureModuleExists` function to `getSpartacusFeatureModule`, where it returns the source else return undefined. 

As I see, it is being used in line 300 for the `checkAppStructure` to verify  if it is true. It won't affect the condition since it returns an object instead of undefined. Therefore, we can just re-use it here instead I think

Reply:
Good catch! Done.